src/AppInstallerCommonCore/Public/winget/ManifestCommon.h

```diff
@@ -112,20 +112,58 @@ namespace AppInstaller::Manifest
         Preview,
     };
 
-    struct PackageDependency
+    enum class DependencyType
     {
-        string_t Id;
-        string_t MinVersion;
+        WindowsFeature,
+        WindowsLibraries,
+        Package,
+        External
     };
 
     struct Dependency
     {
-        std::vector<string_t> WindowsFeatures;
-        std::vector<string_t> WindowsLibraries;
-        std::vector<PackageDependency> PackageDependencies;
-        std::vector<string_t> ExternalDependencies;
+        DependencyType Type;
+        string_t Id;
+        std::optional<string_t> MinVersion;
+
+        Dependency(DependencyType type, string_t id, std::optional<string_t> minVersion) : Type(type), Id(id), MinVersion(minVersion) {}
+        Dependency(DependencyType type, string_t id) : Type(type), Id(id) {}
+        Dependency(DependencyType type) : Type(type) {}
     };
 
+    struct DependencyList
+    {
+        std::vector<Dependency> dependencies;
+
+        DependencyList() {}
+
+        void Add(Dependency dependency) { dependencies.push_back(dependency); }

```

Comment:
`const Dependency&` and/or `Dependency&&`.  Probably both overloads are worthwhile.

Reply:
This should also be checking for validity; and probably in different configurations:
1. Manifest validation would require uniqueness of dependencies; so error if a second PackageDependency shows up for the same package name.
2. Relaxed mode would be used for things like `import` and loading manifests (not for validation).  It would not error on a second PackageDependency, but rather take the higher minimum version.

Reply:
Is there some configuration I can check or should I just use a parameter to choose between each?

Reply:
The list object should take a parameter to choose; the only place that it would be set to use strict mode would be the validate command (aka the `fullValidation` bool is set).

Reply:
I think as for this step (show/report dependencies) it will be better to just output the dependencies to the user. Although merging dependencies, i.e., take the higher min version if the dependency exists, is a nice idea to have now.